mmengine/runner/runner.py

```diff
@@ -56,6 +56,35 @@
 OptimWrapperType = Union[OptimWrapper, OptimWrapperDict]
 
 
+class SlicedDataset:
+
+    def __init__(self, dataset, length) -> None:
+        self.dataset = dataset
+        self.length = length
+        assert self.length <= len(self.dataset)
+        self.count = 0
+        self.dataset_iter = None
```

Comment:
We should turn them into private variables to prevent both the original dataset and the sliced dataset from containing the same fields. Users could mistakenly access attributes of the sliced dataset when trying to access properties of the original dataset.